Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,125 @@
+import PackagePlugin
+import Foundation
+
+enum PluginError: Swift.Error, CustomStringConvertible, LocalizedError {
+    case incompatibleTarget(targetName: String)
+    case noTargetOrDependenciesWithExpectedFiles(targetName: String, dependencyNames: [String])
+    case badArguments([String])
+    case noTargetsMatchingTargetName(targetName: String)
+    case tooManyTargetsMatchingTargetName(targetName: String, matchingTargetNames: [String])
+    case fileErrors([FileError])
+
+    var description: String {
+        switch self {
+        case .incompatibleTarget(let targetName):
+            return "Incompatible target called '\(targetName)'. Only Swift source targets can be used with the Swift OpenAPI Generator plugin."
+        case .noTargetOrDependenciesWithExpectedFiles(let targetName, let dependencyNames):
+            let introduction = dependencyNames.isEmpty ? "Target called '\(targetName)' doesn't contain" : "Target called '\(targetName)' or its local dependencies \(dependencyNames) don't contain"
+            return "\(introduction) any config or document files with expected names. For OpenAPI code generation, a target needs to contain a config file named 'openapi-generator-config.yaml' or 'openapi-generator-config.yml', as well as an OpenAPI document named 'openapi.yaml', 'openapi.yml' or 'openapi.json' under target's source directory. See documentation for details."
+        case .badArguments(let arguments):
+            return "On Xcode, use Xcode's command plugin UI to choose one specific target before hitting 'Run'. On CLI make sure arguments are exactly of form '--target <target-name>'. The reason for this error is unexpected arguments: \(arguments)"
+        case .noTargetsMatchingTargetName(let targetName):
+            return "Found no targets matching target name '\(targetName)'. Please make sure the target name argument leads to one and only one target."
+        case .tooManyTargetsMatchingTargetName(let targetName, let matchingTargetNames):
+            return "Found too many targets matching target name '\(targetName)': \(matchingTargetNames). Please make sure the target name argument leads to a unique target."
+        case .fileErrors(let errors):
+            return "Found file errors: \(errors)."
+        }
+    }
+
+    var errorDescription: String? {
+        description
+    }
+
+    /// The error is definitely due to misconfiguration of a target.
+    var isDefiniteMisconfigurationError: Bool {
```

Comment:
```suggestion
    var isMisconfigurationError: Bool {
```

You can drop the "Definite", as `isMisconfigurationError` already communicates what you need.